spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```
@@ -75,32 +82,37 @@ public FirestoreTemplate(FirestoreStub firestore, String parent) {
 		this.databasePath = parent.substring(0, StringUtils.ordinalIndexOf(parent, "/", 4));
 	}
 
-	public <T> Mono<T> findById(Publisher idPublisher, Class<T> aClass) {
-		return findAllById(idPublisher, aClass).next();
+	public <T> Mono<Boolean> existsById(Publisher<String> idPublisher, Class<T> entityClass) {
+		return Flux.from(idPublisher).next()
+				.flatMap(id -> getDocument(id, entityClass, NAME_ONLY_MASK))
+				.map(d -> d == DocumentReadContext.EMPTY ? Boolean.FALSE : Boolean.TRUE)
+				//.onErrorReturn(throwable -> throwable.getMessage().startsWith("NOT_FOUND: Document"), Boolean.FALSE)
+				.onErrorMap(
+						throwable -> new FirestoreDataException("Unable to test for document existence", throwable));
+	}
+
+	public <T> Mono<T> findById(Publisher idPublisher, Class<T> entityClass) {
+		return findAllById(idPublisher, entityClass).next();
```

Should we raise an error if more than one entity is found?